packages/appkit/src/client.ts

```diff
@@ -1223,6 +1240,9 @@ export class AppKit {
       await provider.getSmartAccountEnabledNetworks()
       this.setLoading(false)
     })
+    provider.onSocialConnected(({ userName }) => {
+      this.setUsername(userName, ChainController.state.activeChain as ChainNamespace)
+    })
```

Comment:
We can only get the username when connecting to a social. This means we'd have to store the username in local storage like we currently do and use that value in our `AccountController.state.username` state.

Magic doesn't give us back username when calling `this.magic.getUser()` in secure site

Reply:
wym? wouldn't `connect` have the username when reconnecting?

Reply:
For some reason it doesn't expose username. It only exposes username when connecting to a social